api/userstore/client.go

```diff
@@ -16,11 +16,31 @@ type UserStore struct {
 	api restapi.Connector
 }
 
+type usersResult struct {
+	Count int    `json:"count"`
+	Items []User `json:"items"`
+}
+
 // New creates a new user-store client instance
 func New(api restapi.Connector) *UserStore {
 	return &UserStore{api: api}
 }
 
+// GetLocalUsers get a local user with details
+func (store *UserStore) GetLocalUsers(filters ...string) ([]User, error) {
```

Comment:
I would suggest to define some kind of type to describe pagination and filters. It is impossible to understand what requirement and what position to make a call.

Reply:
Would be that sufficient enough: 

```
func (store *UserStore) GetLocalUsers(offset, limit, userID, username string) ([]User, error) {
	result := usersResult{}

	_, err := store.api.
		URL("/local-user-store/api/v1/users?offset=%s&limit=%s&username=%s&id=%s", offset, limit, username, userID).
		Get(&result)

	return result.Items, err
}
```

Or do you mean to create an actual type struct for the pagination and filters?

Reply:
This depends what you want to deliver from developer perspective.

1. Please keep-in mind there is Query function that simplify the delivery of parameters. This function handles spacing, etc
https://github.com/SSHcom/privx-sdk-go/blob/master/restapi/client.go#L115

```go
type Params struct {
  UserID string `json:"id,omitempty"`
  Username string `json:"username,omitempty"`
 ...
}

store.api.
  URL("/local-user-store/api/v1/users").
  Query(params)
```
 
2. The proposed implementation do not escape values or properly handles empty values
```
GetLocalUsers("", "", "a&foo=bar", )
```

3. How is it typical to define offset and limits in the requests. Are you going to add more and more functions like this?
May be you need to simplify a life for developers using the following constrains. It is really up to you

```go
type Params struct {
   Offset string
   Limit string
}

type FilterUser struct {
   Params
   UserID string
   Username string
}
```

Reply:
Thank you for the very detailed answer! 
I'm gonna use the offset and limit only for 2 endpoints inside the user store, which is still I guess a lazy excuse not to implement the Params and FilterUser struct, to make the SDK like you said more developer friendly in the future. 

Thank you for pointing me out the `Query` function, totally missed that one! 
One question regarding this function, it is accepting an empty interface{}, which takes any type if understood it correctly, right?  What would be the best way in your opinon to pass those 4 filter parameters dynamically to the Query function? 

Sorry for my little knowledge around golang, specilly the interface type is still a bit confusing for me.

Reply:
Rough approximation, would look like this:

```go
func (store *UserStore) GetLocalUsers(offset, limit, userID, username string) ([]User, error) {
	result := usersResult{}
       params := struct {
           Offset string `json:"offset,omitempty"`
           Limit string `json:"limit,omitempty"`
           ...
       } {
          Offset: offset,
          Limit: limit,
          ...
       }

	_, err := store.api.
		URL("/local-user-store/api/v1/users).
                Query(&params).
		Get(&result)

	return result.Items, err
}
```